### DISCUSSION TITLE
민감정보 관리 방법에 대해

### DISCUSSION THREAD

**Discussion by amaran-th** [BDD-CLUB/01-doo-re-back#44] [논의 사항]
안녕하세요~ 1/24(수)에 잠시 논의했던 안건입니다!
현재까지 구현된 인프라 구성에 따르면, 로컬 환경에서 사용하는 yml 파일과 실제 dev 환경에서 사용할 yml 파일을 분리해서 적용하고 있는데요, 이 dev yml 파일을 어떻게 관리할지 논의하고 싶습니다!
1. dev yml 파일을 gitignore 처리를 해서, yml 파일에 변경사항이 생길 때마다 서버에 덮어쓰기한다.
2. Github Secret에 민감 정보 데이터를 저장한다.
3. dev yml 파일을 암호화해서 올린다.(jasypt)
4. 서브모듈로 관리한다.

이 외에 다른 아이디어가 있다면 공유 부탁드립니다!

제대로 정리한 건 아니지만 참고하실 수 있게 옛날에 기록해두었던 글 공유합니다!
[블로그 글](https://amaran-th.github.io/%EC%A3%BC%EC%A0%80%EB%A6%AC%EC%A3%BC%EC%A0%80%EB%A6%AC/Kerdy%20%ED%94%84%EB%A1%9C%EC%A0%9D%ED%8A%B8%EC%9D%98%20%EC%9D%B8%ED%94%84%EB%9D%BC%EC%97%90%20%EB%8C%80%ED%95%9C%20%EB%85%BC%EC%9D%98/)

**Comment by shkisme** [comment]
저는 1번 방식으로 오랜 기간 프로젝트를 진행해봤었는데요, 다음과 같은 문제점들이 있었습니다.
- yml 파일 동기화가 너무 어려웠다. (업데이트 될 때마다 일일히 보내줘야 함 + 에러난다고 연락오는 경우 엄청 잦았는데 대부분 비최신 yml 문제였음)
- 새로운 팀원이 추가될 때 마다 yml을 매번 보내줘야 한다는 불편함이 있었다.
- yml 파일을 받은 팀원이 사용할 때 파일 위치를 실수로 git ignore가 설정되지 않은 경로로 잘못 위치시켜 깃허브에 노출된 적이 있었다.
- 받은 yml 파일을 이동 & 복사하는 과정에서 문제가 생겨 적용이 안되는 등의 문제가 있었다.

장점은, 추가적인 설정을 할 필요가 없어서 간단하다. 정도가 있겠네요 😅

2번의 경우, 다시 조회할 수 없기에 결국은 어딘가에 키값을 저장해두고 공유해야하는 단점이 있겠네요.

3, 4번의 경우 경험해보지 못해서 공유해주신 글을 확인해보았는데요, 3번 방식은 Github secret으로 관리해야 할 포인트가 하나로 줄어서 간편해보이고, 서브모듈로 관리하는 방법도 편리해보이네요. 다만, 제가 보기에는 복잡한 추가적인 설정이 필요해 보이네요.

저는 이렇게 생각하지만, 이 부분은 현재의 인프라 구성에 많은 기여를 하신 @02ggang9 님의 의견이 중요해보입니다~! 😃

**Comment by JJimini** [comment]
저도 처음에는 3번이 괜찮다고 느꼈는데요, 글을 보니 결국 암호화가 되었지만 공개되는 건 똑같다는 단점이 걸리네요. 저도 사용해본적은 없지만 4번 방식을 경험해보는 것도 좋을 것 같아요

**Comment by 02ggang9** [comment]
어제 결정된 방법은 공개해도 되는 yml과 민감한 정보가 담긴 yml을 분리하고 민감한 정보가 담긴 yml(local, dev, prod)을 서브 모듈로 관리하기로 했습니다.
